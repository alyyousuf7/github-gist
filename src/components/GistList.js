import PropTypes from 'prop-types';

function GistList({ children, onClickNext, onClickPrev }) {
  return <div>
    <input type="button" value ="prev" onClick={onClickPrev} />
    {children.length > 0 && <input type="button" value ="next" onClick={onClickNext} />}
    <ul className="list-gist">
      {children.map((gist, i) => <li key={i} className="list-item-gist">
        {gist}
      </li>)}
    </ul>
  </div>
}

GistList.propTypes = {
  children: PropTypes.arrayOf(PropTypes.node).isRequired,
  onClickNext: PropTypes.func.isRequired,
  onClickPrev: PropTypes.func.isRequired,
};

export default GistList;
