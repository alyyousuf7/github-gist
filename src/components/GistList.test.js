
import { render } from '@testing-library/react';

import GistList from './GistList';

describe('GistList', () => {
  it('should render correctly', async () => {
    const { container } = render(<GistList onClickPrev={() => {}} onClickNext={() => {}}>
      <div>child 1</div>
      <div>child 2</div>
    </GistList>);

    expect(container).toMatchSnapshot();
  });
});
