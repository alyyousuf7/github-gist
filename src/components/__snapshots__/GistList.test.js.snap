// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GistList should render correctly 1`] = `
<div>
  <div>
    <input
      type="button"
      value="prev"
    />
    <input
      type="button"
      value="next"
    />
    <ul
      class="list-gist"
    >
      <li
        class="list-item-gist"
      >
        <div>
          child 1
        </div>
      </li>
      <li
        class="list-item-gist"
      >
        <div>
          child 2
        </div>
      </li>
    </ul>
  </div>
</div>
`;
